source/agora/node/Registry.d

```diff
@@ -872,6 +875,11 @@ private struct ZoneData
 
         // Address loop consumes data, gather following first
         const TYPE node_type = to!TYPE(results.front["type"].as!ushort);
+
+        // Check query type; QTYPE is superset of TYPE, casting is OK
+        if (type != QTYPE.ALL && type != cast(QTYPE) node_type)
+            return TypedPayload.init;
+
```

Comment:
This is not quite correct though:

> CNAME RRs cause special action in DNS software.  When a name server
fails to find a desired RR in the resource set associated with the
domain name, it checks to see if the resource set consists of a CNAME
record with a matching class.  If so, the name server includes the CNAME
record in the response and restarts the query at the domain name
specified in the data field of the CNAME record.  The one exception to
this rule is that queries which match the CNAME type are not restarted.

[RFC1034, 3.6.2. Aliases and canonical names](https://datatracker.ietf.org/doc/html/rfc1034#section-3.6.2)